backend/app/anti_recommenders/open_ai/open_ai_anti_recommender.py

```diff
@@ -32,19 +31,23 @@ def _create_query(self, record_key: str) -> str:
                  Give each answer on a new line, and in the format: Number - Title - URL."
         )
 
-    @abstractmethod
-    def _build_chain(self) -> None:
-        pass
+    def _generate_anti_recommendendations(  # noqa: PLR0913
+        self,
+        record_key: str,
+        callable_to_build_chain: Callable,
```

Comment:
These need to have more refined types.

Reply:
You don't need to name them with `callable_to`, you can just say `build_chain` since it will be apparent what the callable does from the type signature.

Reply:
I have fixed the names of the callables.